### PR TITLE
MM-47105 :  Migrate "components/more_direct_channels/list_item/list_item.test.jsx" to Typescript

### PR DESCRIPTION
**Pull request by cannalee90**
**Summary**

migrate list_item.test.tsx to typescript

**Ticket Link**

Fixes https://github.com/mattermost/mattermost-server/issues/21132
Fixes https://mattermost.atlassian.net/browse/MM-47105

Related Pull Requeests

NA

Screenshots

NA

Release Note

NA

### CODE REVIEW

File: components/more_direct_channels/list_item/list_item.test.tsx [hunk 1]
```diff
@@ -6,6 +6,8 @@ import React from 'react';
 
 import {General} from 'mattermost-redux/constants';
 
+import {OptionValue} from '../types';
+
 import ListItem from './list_item';
 
 describe('ListItem', () => {
```
Comment (nevyangelova) [inline components/more_direct_channels/list_item/list_item.test.tsx:13]: Thanks @cannalee90 can you please add a type for the props?

**Review by nevyangelova — CHANGES_REQUESTED** [review]

File: components/more_direct_channels/list_item/list_item.test.tsx (hunk 1 above)
Comment (cannalee90) [reply components/more_direct_channels/list_item/list_item.test.tsx:13]: no problem. I add the props and apply it

**Review by M-ZubairAhmed — APPROVED** [review]
🆗  It looks good to me. Thank you

File: components/more_direct_channels/list_item/list_item.test.tsx (hunk 1 above)
Comment (cannalee90) [reply components/more_direct_channels/list_item/list_item.test.tsx:13]: @nevyangelova Hi. Would you resolve this issue?

**Review by nevyangelova — APPROVED** [review]
LGTM :)

**Pull request merged**